test/integration/admin_dashboard_test.rb

```diff
@@ -15,16 +15,31 @@ class AdminDashboardTest < ActionDispatch::IntegrationTest
     assert page.has_content?("Admin Dashboard")
   end
 
-  test "Admin sees content on admin dashboard" do
-    skip
-    # As an Admin
-    # When I visit "/admin/dashboard"
-    # I will see a heading on the page that says "Admin Dashboard"
-    # As a registered user
-    # When I visit "/admin/dashboard"
-    # I get a 404
-    # As an unregistered user
-    # When I visit "/admin/dashboard"
-    # I get a 404
+  test "user cannot access admin dashboard" do
+    User.create(username: "cole", name: "Cole Hall", password: "password", role: 0)
+
+    visit root_path
+
+    click_button "Login"
+    fill_in "Username", with: "cole"
+    fill_in "Password", with: "password"
+    click_button "Login"
+
+    assert page.has_content?("Welcome, Cole Hall!")
+    assert '/dashboard', current_path
+
+    visit admin_dashboard_path
+save_and_open_page
+    assert page.has_content?("404")
+  end
+
+  test "unregistered user cannot access admin dashboard" do
+    visit root_path
+
+    assert page.has_content?("Login")
+
+    visit '/admin/dashboard'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.